### PR TITLE
Feature/day two

### PR DESCRIPTION
**Pull request by SamTush**
## What does this PR do?

This PR is raised to implement the following features: 👍 :

- [x]  Add species column to animals table
- [x] Insert more data to the table 

## Description of the task to be completed ↓↓ 

- Insert more data to the table

## Screenshots of the queries.

- Update the animals table by setting the species column to digimon for all animals that have a name ending in mon.

Screenshot 2023-05-11 at 14 59 00

- Update the animals table by setting the species column to pokemon for all animals that don't have species already set.

Screenshot 2023-05-11 at 15 00 39

- Transaction results 
Screenshot 2023-05-11 at 21 15 01

- How many animals are there?
Screenshot 2023-05-11 at 15 29 52

- How many animals have never tried to escape?
Screenshot 2023-05-11 at 15 32 53

- What is the average weight of animals?

Screenshot 2023-05-11 at 21 15 41

- Who escapes the most, neutered or not neutered animals?

Screenshot 2023-05-11 at 21 15 46

- What is the minimum and maximum weight of each type of animal?

Screenshot 2023-05-11 at 21 15 53

- What is the average number of escape attempts per animal type of those born between 1990 and 2000?

Screenshot 2023-05-11 at 16 25 53

#### Thanks for reviewing my PR 😊

### CODE REVIEW

File: queries.sql [hunk 1]
```diff
@@ -1,3 +1,46 @@
...
+
+
+-- transactions
+
+BEGIN;
+DELETE FROM animals WHERE date_of_birth > '2022-01-01';
+SAVEPOINT weight_change;
+UPDATE animals SET weight_kg = weight_kg * -1;
+ROLLBACK TO weight_change;
+COMMIT;
```
Comment (Deepakdanger) [inline queries.sql]: - [ ] after `rollback` & before `commit` we have to `UPDATE` all animals' weights that are negative to be their weight multiplied by -1.
- `UPDATE animals SET weight_kg = weight_kg * -1 WHERE weight_kg < 0;`

File: queries.sql [hunk 2]
```diff
@@ -1,3 +1,46 @@
...
+
+SELECT COUNT(*) FROM animals;
+SELECT COUNT(*) FROM animals WHERE escape_attempts = 0;
+SELECT AVG(weight_kg) FROM animals;
+SELECT * FROM animals ORDER BY escape_attempts DESC LIMIT 1;
```
Comment (Deepakdanger) [inline queries.sql]: - [ ] Here we have to find `Who escapes the most, neutered or not neutered animals?`.
```sql
SELECT neutered, AVG(escape_attempts) FROM animals
   GROUP BY neutered;
```

File: queries.sql [hunk 3]
```diff
@@ -1,3 +1,46 @@
...
+WHERE neutered = 't';
+
+SELECT * FROM animals
+WHERE name <> 'Gabumon';
+
+SELECT * FROM animals
+WHERE weight_kg >= 10.4 AND weight_kg <=17.3;
+
+
+-- transactions
```
Comment (Deepakdanger) [inline queries.sql:32]: - You have missed many required transaction,kindly add those.
- [ ] `UPDATE` the animals table by setting the species column to unspecified, then roll back the change.

```sql
   BEGIN; -- start transaction

   UPDATE animals
   SET species = 'unspecified'; -- make change

   SELECT species from animals; -- verify that change was made
   ROLLBACK;

   SELECT species from animals;
```
- [ ] `UPDATE` the animals table by setting the species column to digimon for all animals that have a name ending in mon. & then `UPDATE` the animals table by setting the species column to pokemon for all animals that don't have species already set. & Commit the transaction.

```sql
   BEGIN;
   UPDATE animals SET species = 'digimon' WHERE name LIKE '%mon'; 
   UPDATE animals SET species = 'pokemon' WHERE species IS NULL; 

   SELECT species from animals; -- verify that change was made
   COMMIT;
   SELECT species from animals;
```
- [ ] delete all records in the animals table, then roll back the transaction.

**Review by Deepakdanger — CHANGES_REQUESTED** [review]
Hi @SamTush , 

Good job so far! 👏
There are some issues that you still need to work on to go to the next project but you are almost there!

## To Highlight :100: 
- No linter error :heavy_check_mark: 
- Website looks professional :+1: 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by SamTush** [comment]
Thank you @Deepakdanger  for reviewing my PR👍🏽 Cheers and happy coding😎

**Review by KanzaTahreem — APPROVED** [review]
# Approved 🥳

Hi @SamTush 🙋‍♀️

Great work implementing the changes suggested by the previous reviewer 💪 **_Your project is COMPLETE! There is nothing else to say other than... it's time to merge it! 🚀 Congratulations!_**



## Highlights 🎯
There are some excellent points that you have made perfectly. Keep up the great work! 👏

-  ✅ Great work using the right Gitflow 
-  ✅ Commits are descriptive, good work.
-  ✅ Good PR description 

## Optional suggestions ❕
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

- [OPTIONAL] Please make sure to customize your README according to the project on which you are working on to make it more professional 🤓⛏

Cheers and Happy coding!👏👨‍💻💻

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by SamTush** [comment]
Thank you @KanzaTahreem for reviewing my PR👍🏽 Cheers and happy coding😎

**Pull request merged**